test/unit/mdc-dialog/foundation.test.js

```diff
@@ -83,92 +82,76 @@ test('#open registers all events registered within open()', () => {
   td.verify(mockAdapter.registerSurfaceInteractionHandler('click', td.matchers.isA(Function)));
   td.verify(mockAdapter.registerDocumentKeydownHandler(td.matchers.isA(Function)));
   td.verify(mockAdapter.registerInteractionHandler('click', td.matchers.isA(Function)));
-  td.verify(mockAdapter.registerFocusTrappingHandler(td.matchers.isA(Function)));
 });
 
 test('#close deregisters all events registered within open()', () => {
   const {foundation, mockAdapter} = setupTest();
 
-  foundation.open();
   foundation.close();
 
   td.verify(mockAdapter.deregisterSurfaceInteractionHandler('click', td.matchers.isA(Function)));
   td.verify(mockAdapter.deregisterDocumentKeydownHandler(td.matchers.isA(Function)));
   td.verify(mockAdapter.deregisterInteractionHandler('click', td.matchers.isA(Function)));
-  td.verify(mockAdapter.deregisterFocusTrappingHandler(td.matchers.isA(Function)));
 });
 
-test('#open adds the open class to reveal the dialog', () => {
+test('#open makes the dialog visible', () => {
   const {foundation, mockAdapter} = setupTest();
 
   foundation.open();
-  td.verify(mockAdapter.addClass(cssClasses.OPEN));
+
+  td.verify(mockAdapter.setStyle('visibility', 'visible'));
 });
 
-test('#close removes the open class to hide the dialog', () => {
+test('#close makes the dialog hidden', () => {
   const {foundation, mockAdapter} = setupTest();
 
   foundation.close();
-  td.verify(mockAdapter.removeClass(cssClasses.OPEN));
+
+  td.verify(mockAdapter.setStyle('visibility', 'hidden'));
 });
 
-test('#open adds scroll lock class to the body', () => {
+test('#open adds the open class to reveal the dialog', () => {
   const {foundation, mockAdapter} = setupTest();
 
   foundation.open();
-  td.verify(mockAdapter.addBodyClass(cssClasses.SCROLL_LOCK));
+  td.verify(mockAdapter.addClass(cssClasses.OPEN));
 });
 
-test('#close removes the scroll lock class from dialog background', () => {
+test('#close removes the open class to hide the dialog', () => {
   const {foundation, mockAdapter} = setupTest();
 
   foundation.close();
-  td.verify(mockAdapter.removeBodyClass(cssClasses.SCROLL_LOCK));
+  td.verify(mockAdapter.removeClass(cssClasses.OPEN));
 });
 
-test('#open makes elements tabbable', () => {
+test('#open adds scroll lock class to the body', () => {
   const {foundation, mockAdapter} = setupTest();
-  td.when(mockAdapter.hasClass(cssClasses.OPEN)).thenReturn(false);
-  td.when(mockAdapter.getFocusableElements()).thenReturn(['foo', 'bar']);
 
-  foundation.init();
   foundation.open();
-  td.verify(mockAdapter.restoreElementTabState('foo'));
-  td.verify(mockAdapter.restoreElementTabState('bar'));
+  td.verify(mockAdapter.addBodyClass(cssClasses.SCROLL_LOCK));
 });
 
-test('#close makes elements untabbable', () => {
+test('#close removes the scroll lock class from dialog background', () => {
```

Comment:
This is on me for coming up with the naming convention for the component, but potentially we should rephrase "background" to "content underneath dialog" for the test description. Aria calls things behind the dialog "background" which is why I called it this to begin with, but I feel this could too easily be confused with the concept of a dialog backdrop, which we also have.